File: 0283-move-zeroes/0283-move-zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        l, r = -1, 0
        while r < len(nums):
            if nums[r] != 0:
                l += 1
                nums[l] = nums[r]
            r += 1
        
        for i in range(l+1, len(nums)):
            nums[i] = 0
        
        return nums

File: 0283-move-zeroes/0283-move-zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        int i = 0, j = 0;
        while (i < nums.length && j < nums.length) {
            while (i < nums.length && nums[i] != 0) i++;
            while (j < nums.length && nums[j] == 0) j++;
            if (i < nums.length && j < nums.length) {
                if (i < j) {
                    int curJ = nums[j];
                    nums[j] = 0;
                    nums[i] = curJ;
                } else {
                    j = i;
                }
            }
        }
        
    }
}